src/common/transformations/src/transformations/common_optimizations/transpose_sinking_interpolate.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/transpose_sinking_interpolate.hpp"
+
+#include <openvino/pass/pattern/op/or.hpp>
+
+#include "itt.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+#include "transformations/common_optimizations/transpose_sinking_utils.hpp"
+#include "transformations/rt_info/transpose_sinking_attr.hpp"
+
+using namespace ov;
+using namespace ov::opset10;
+using namespace ov::pass::pattern;
+using namespace transpose_sinking;
+
+ov::pass::TransposeSinkingInterpolateForward::TransposeSinkingInterpolateForward() {
+    MATCHER_SCOPE(TransposeSinkingInterpolateForward);
+    auto const_label = wrap_type<Constant>();
+    auto transpose_label = wrap_type<Transpose>({any_input(), const_label});
+    auto main_node_label = wrap_type<Interpolate>({transpose_label, any_input(), any_input(), any_input()});
+
+    matcher_pass_callback matcher_pass_callback = [=](Matcher& m) {
+        const auto& pattern_to_node = m.get_pattern_map();
+
+        auto& main_node = pattern_to_node.at(main_node_label);
+        auto transpose = std::dynamic_pointer_cast<Transpose>(pattern_to_node.at(transpose_label));
+        if (!transpose) {
+            return false;
+        }
+
+        auto transpose_const = as_type_ptr<Constant>(pattern_to_node.at(const_label));
+        if (!transpose_const) {
+            return false;
+        }
+
+        // remove Transpose on 1st input:
+        auto transpose_parent = main_node->input_value(0).get_node()->input_value(0);
```

Comment:
can it be a case that `main_node->input_value(0).get_node()` is a node that does not have any input like `Parameter` or `Const`

Reply:
this is Transpose node,  fixed https://github.com/openvinotoolkit/openvino/pull/16283